app/com/arpnetworking/metrics/portal/scheduling/impl/JobScheduler.java

```diff
@@ -0,0 +1,225 @@
+ */
+public final class JobScheduler<T> extends AbstractActorWithTimers {
+
+    private final JobRef<T> _jobRef;
+    private final Clock _clock;
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param jobRef The job to intermittently execute.
+     * @return A new props to create this actor.
+     */
+    public static <T> Props props(final JobRef<T> jobRef) {
+        return props(jobRef, Clock.systemUTC());
+    }
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param jobRef The job to intermittently execute.
+     * @param clock The clock the scheduler will use, when it ticks, to determine whether it's time to run the next job(s) yet.
+     * @return A new props to create this actor.
+     */
+    protected static <T> Props props(final JobRef<T> jobRef, final Clock clock) {
+        return Props.create(JobScheduler.class, () -> new JobScheduler<>(jobRef, clock));
+    }
+
+    private JobScheduler(final JobRef<T> jobRef, final Clock clock) {
+        _jobRef = jobRef;
+        _clock = clock;
+        timers().startPeriodicTimer("TICK", Tick.INSTANCE, TICK_INTERVAL);
+    }
+
+    /**
+     * Computes the time we should schedule an extra tick for, if we'll need an extra tick before the next naturally occurring one.
+     *
+     * @param now The current time.
+     * @param timeToAwaken The time we want to wake up right after.
+     * @return The time until we should wake up, if that's before the next tick; else {@code empty}.
+     */
+    protected static Optional<FiniteDuration> timeUntilExtraTick(final Instant now, final Instant timeToAwaken) {
+        final FiniteDuration delta = Duration.fromNanos(ChronoUnit.NANOS.between(now, timeToAwaken));
+        if (delta.lt(TICK_INTERVAL)) {
+            return Optional.of(delta.plus(SLEEP_SLOP));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .match(Tick.class, message -> {
+                    final Optional<Job<T>> maybeJob = _jobRef.get();
+                    if (!maybeJob.isPresent()) {
+                        LOGGER.error()
+                                .setMessage("no such job")
+                                .addData("jobRef", _jobRef)
+                                .log();
+                        return;
+                    }
+                    final Job<T> job = maybeJob.get();
+                    final Optional<Instant> maybeNextRun = job.getSchedule().nextRun(_jobRef.getLastRun());
+                    if (!maybeNextRun.isPresent()) {
+                        getSelf().tell(PoisonPill.getInstance(), getSelf());
+                        return;
+                    }
+                    final Instant nextRun = maybeNextRun.get();
```

Comment:
At this point you may want a helper method that takes a job and scheduled instant and does the "next run logic".